### PR TITLE
Add resilience to NAT rebinding as a property

### LINKED ISSUE #9: Mention NAT rebinding
This was brought up by Ian, but basically the idea is that the mechanism should be resilient to a NAT rebind that keeps the flow on the same logical CSP, rather than breaking.

Relatedly, it should be clear that switching operators will break the communication channel due to the on-path nature of the communication.

### PR DESCRIPTION
**Pull request by SpencerDawkins**
closes #9

### CODE REVIEW

**Review by fluffy — COMMENTED** [review]
I worry that this is magnet for a whole bunch of weird arguments and we don't need it in the charter one way or the other.  I will point out most the boxes I am aware of that do rate limiting today, do not track across a change in source IP port.

**Comment by danwing** [comment]
SCONE isn't doing rate limiting, though; rather, it is informing the client of the network's preferred rate for that subscriber (or more generally, that "class" of subscriber: gold/silver/bronze).  This network's rate policy is the same even if the client's source IP address changes; the network's rate policy only changes if the subscriber upgrades/downgrades to a different class.

**Comment by fluffy** [comment]
I get what you are saying and that makes sense but I worry about the following. Our security properties are probably going to say we only trust this information if the box can prove that it is in a position where it could shape traffic. So if we want to extend that to all the source ports, then we are in a place where for this to be true, we have to assume no source based routing. However having the source based routing is pretty common for load distribution.  So I worry by taking on this requirement, we might make it harder to make our security guarantees.  

I have no idea if the protocol will work like this at all and this seems like a good thing for the WG to sort out once it gets going, it just seemed like a level of detail that did not need to be in the charter. Or to put another way, I was not understanding why this needed to be in the charter.

**Comment by SpencerDawkins** [comment]
Hi, @fluffy,

> I get what you are saying and that makes sense but I worry about the following. Our security properties are probably going to say we only trust this information if the box can prove that it is in a position where it could shape traffic. So if we want to extend that to all the source ports, then we are in a place where for this to be true, we have to assume no source based routing. However having the source based routing is pretty common for load distribution. So I worry by taking on this requirement, we might make it harder to make our security guarantees.

That helps me understand what you're thinking. I'm leaning towards what @danwing is thinking, mostly because I wasn't expecting boxes that do rate limiting to have high confidence that a flow with a different port number and IP address is the same flow it's already aware of. But, 
 
> I have no idea if the protocol will work like this at all and this seems like a good thing for the WG to sort out once it gets going, **it just seemed like a level of detail that did not need to be in the charter. Or to put another way, I was not understanding why this needed to be in the charter.**

is **EXACTLY** what we need to talk about, so we know whether to merge the PR, change it, or abandon it and change the label from "charter" to "InADocument".

**Review by ihlar — APPROVED** [review]

File: documents/charter.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ That is, no off-path element is needed to establish the communication channel be
 1. Optionality.
 The communication channel is strictly optional for the functioning of application flows.
 A client's application flow must function even if the client does not establish the channel.
+1. Resiliant to NAT rebinding. 
+The mechanism will allow the communication channel to be resiliant to NAT rebinding, as long as the client is still served by the same logical CSP.
```
Comment (martinthomson) [inline documents/charter.md:34]: CSP is not on the acronym list, so it needs to be spelled out.

**Pull request merged**